### PR TITLE
FSPT-598: Access grant funding 'grant' page

### PR DESCRIPTION
**Pull request by samuelhwilliams**
Relies on https://github.com/communitiesuk/funding-service/pull/418

## 🎫 Ticket
https://mhclgdigital.atlassian.net/browse/FSPT-598

## 📝 Description
Builds out a basic details page for a grant, from the view of an Access grant funding user.

This is just a developer draft of the page, aimed at proving some part of the end-to-end funding journey, so will not look like this when real users access it.

The page just lists all of the collections created for the grant on the page, and in the future will let the user create/edit/manage a submission against any of those collections.

This is a fully-public page; if you aren't logged in, you can still see the page. You won't see any collections and will instead see a link to the sign in with a magic link page.

## 📸 Show the thing (screenshots, gifs)
image

## 🧪 Testing
I loaded the page as a platform admin, and in a private window (ie anonymous user).

### Review Readiness
- [x] PR title and description provide sufficient context for reviewers
- [x] Commits are logical, self-contained, and have clear descriptions

### Testing
- [x] I have tested this change and it meets the acceptance criteria for the ticket
~- [ ] I need the reviewer(s) to pull and run this change locally~
~- [ ] New (non-developer) functionality has appropriate unit and integration tests~
~- [ ] End-to-end tests have been updated (if applicable)~
~- [ ] Edge cases and error conditions are tested~

### CODE REVIEW

File: app/developers/templates/developers/access/grant_details.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+{% set page_title = "Grants" %}
+
+{% block beforeContent %}
+  {{
+    govukBackLink({
+        "text": "Back",
+        "href": url_for("developers.access.grants_list")
+    })
+  }}
+{% endblock beforeContent %}
```
Comment (MarcUsher) [inline app/developers/templates/developers/access/grant_details.html:16]: Does it make sense for us to have a backlink here if the AGF grant list page is only for platform admins? Seems odd to have it from the point of view of an AGF user, or even an unauthed user visiting this page to be prompted to login.

Comment (samuelhwilliams) [reply app/developers/templates/developers/access/grant_details.html]: Yeah good catch thanks - have made it only visible for platform admins 👀
